src/go/rpk/pkg/cli/transform/build.go

```diff
@@ -140,3 +146,88 @@ func buildRust(ctx context.Context, fs afero.Fs, cfg project.Config, extraArgs [
 	}
 	return nil
 }
+
+const watTemplate = `
+(module
+	(memory (import "js_vm" "memory") 0)
+
+	(func (export "file_length") (result i32)
+          (i32.const %d)
+        )
+
+	(func (export "get_file") (param $buffer_ptr i32)
+		;; Copy the source from data segment 0.
+		(memory.init $source
+			(local.get $buffer_ptr) ;; Memory destination
+			(i32.const 0)           ;; Start index
+			(i32.const %d)          ;; Length
+                )
+		(data.drop $source)
+        )
+	(data $source "%s"))
+`
+
+func createWatFromJavaScript(code []byte) string {
```

Comment:
If possible, can you leave a small comment about the `why` of the encoding/escaping of the code for the wat file?

Reply:
Ah yes, thanks. If there are any other things like this please call them out!